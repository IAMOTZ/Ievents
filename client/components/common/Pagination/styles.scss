@import "../../../styles/variables";

#pagination {
  display: flex;
  justify-content: center;
  margin-top: 50px;
  margin-bottom: 50px;
  .react-paginate {
    ul {
      display: inline-block;
      padding-left: 15px;
      padding-right: 15px;
    }
    li {
      display: inline-block;
      border: 1px solid #d9d9d9;
      padding: 6px 10px 6px 10px;
      &:first-child {
        margin-right: 20px;
      }
      &:last-child {
        margin-left: 20px;
      }
      &.active {
        background-color: $custom-blue;
        border: 1px solid $custom-blue;
        a {
          color: white;
        }
      }
      &.disabled {
        a {
          cursor: default;
          text-decoration: none;
          color: grey;
        }
      }
      a {
        cursor: pointer;
        color: $custom-blue;
        outline: none;
      }
    }
  }
}
