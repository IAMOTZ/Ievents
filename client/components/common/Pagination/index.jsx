import React from 'react';
import PropTypes from 'prop-types';
import ReactPaginate from 'react-paginate';
import './styles.scss';

const Pagination = props => (
  props.pageCount > 1 ?
    <div id="pagination">
      <ReactPaginate
        previousLabel="Previous"
        nextLabel="Next"
        pageCount={props.pageCount}
        marginPagesDisplayed={2}
        pageRangeDisplayed={3}
        onPageChange={props.onPageChange}
        containerClassName="react-paginate"
        activeClassName="active"
      />
    </div> : null
);

Pagination.propTypes = {
  pageCount: PropTypes.number.isRequired,
  onPageChange: PropTypes.func.isRequired
};

export default Pagination;
