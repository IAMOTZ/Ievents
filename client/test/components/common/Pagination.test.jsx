import React from 'react';
import Pagination from '../../../components/common/Pagination';

const props = {
  pageCount: 0,
  onPageChange: jest.fn()
};

const alterProps = newProps => ({ ...props, ...newProps });

describe('<Pagination />', () => {
  describe('Rendering:', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<Pagination {...props} />);
    });
    it('should render correctly when paginationCount is less than one', () => {
      expect(wrapper).toMatchSnapshot();
    });
    it('should render correctly when paginationCount is greater than one', () => {
      wrapper.setProps(alterProps({ pageCount: 5 }));
      expect(wrapper).toMatchSnapshot();
    });
  });
});
