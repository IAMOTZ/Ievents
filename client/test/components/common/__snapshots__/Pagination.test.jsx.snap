// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> Rendering: should render correctly when paginationCount is greater than one 1`] = `
<div
  id="pagination"
>
  <PaginationBoxView
    activeClassName="active"
    breakLabel="..."
    containerClassName="react-paginate"
    disableInitialCallback={false}
    disabledClassName="disabled"
    marginPagesDisplayed={2}
    nextClassName="next"
    nextLabel="Next"
    onPageChange={[MockFunction]}
    pageCount={5}
    pageRangeDisplayed={3}
    previousClassName="previous"
    previousLabel="Previous"
  />
</div>
`;

exports[`<Pagination /> Rendering: should render correctly when paginationCount is less than one 1`] = `""`;
